ios/Codeforces Watcher/Application/features/problems/ProblemsTableViewAdapter.swift

```diff
@@ -13,43 +13,48 @@ import common
 class ProblemsTableViewAdapter: NSObject, UITableViewDelegate, UITableViewDataSource {
     
     var problems: [Problem] = []
+    var visibleProblems: [Problem] = []
```

Comment:
Again, this second array is too much responsibility for a dummy adapter 😉If you really need to store all `problems` locally, store it in VC.